sdk/nodejs/automation/cmd.ts

```diff
@@ -37,26 +46,141 @@ export class CommandResult {
+        return new PulumiCommand(command, version, opts?.root);
+    }
+
+    /**
+     * Installs the Pulumi CLI.
+     *
+     * @param opts.version the version of the CLI. Defaults to the CLI version matching the SDK version.
+     * @param opts.root the directory to install the CLI in. Defaults to $HOME/.pulumi/versions/$VERSION.
+     */
+    static async install(opts?: PulumiCommandOptions): Promise<PulumiCommand> {
+        const optsWithDefaults = withDefaults(opts);
+        try {
+            return await PulumiCommand.get(optsWithDefaults);
+        } catch (err) {
+            // ignore
+        }
+
+        if (process.platform === "win32") {
+            await PulumiCommand.installWindows(optsWithDefaults);
+        } else {
+            await PulumiCommand.installPosix(optsWithDefaults);
+        }
+
+        return await PulumiCommand.get(optsWithDefaults);
+    }
+
+    private static async installWindows(opts: Required<PulumiCommandOptions>): Promise<void> {
+        const response = await got("https://get.pulumi.com/install.ps1");
+        const script = await writeTempFile(response.body, { extension: ".ps1" });
+
+        try {
+            const command = process.env.SystemRoot
+                ? path.join(process.env.SystemRoot, "System32", "WindowsPowerShell", "v1.0", "powershell.exe")
+                : "powershell.exe";
+
+            const args = [
+                "-NoProfile",
+                "-InputFormat",
+                "None",
+                "-ExecutionPolicy",
+                "Bypass",
+                "-File",
+                script.path,
+                "-NoEditPath",
+                "-InstallRoot",
+                opts.root,
+                "-Version",
+                `${opts.version}`,
+            ];
+
+            await exec(command, args);
+        } finally {
+            script.cleanup();
+        }
+    }
+
+    private static async installPosix(opts: Required<PulumiCommandOptions>): Promise<void> {
+        const response = await got("https://get.pulumi.com/install.sh");
+        const script = await writeTempFile(response.body);
+
+        try {
+            const args = [script.path, "--no-edit-path", "--install-root", opts.root, "--version", `${opts.version}`];
+
+            await exec("/bin/sh", args);
+        } finally {
+            script.cleanup();
+        }
+    }
+
+    /** @internal */
+    public run(
+        args: string[],
+        cwd: string,
+        additionalEnv: { [key: string]: string },
+        onOutput?: (data: string) => void,
+    ): Promise<CommandResult> {
+        // all commands should be run in non-interactive mode.
+        // this causes commands to fail rather than prompting for input (and thus hanging indefinitely)
+
+        if (!args.includes("--non-interactive")) {
+            args.push("--non-interactive");
+        }
+
+        // Prepend the installation root to the path to ensure we pickup the
```

Comment:
This doesn't look right? Above we assume the command is `$root/bin/pulumi` but that means PATH here should be `$root/bin`. 
But further more you don't need to amend PATH at all, bundled plugins are already looked up correctly regardless of PATH.

Reply:
The comment is slightly misleading, `prependRootToPath` does add `$root/bin` to `$PATH`, I'll rename the prepend function and update the comment to be more accurate.

I do think we need to fixup the path though. If the user has bundled plugins in $PATH, these will be preferred over the ones next to the executable and we'll potentially get a version mismatch https://github.com/pulumi/pulumi/blob/097fdaa09a1a8b6c76b6aab877fa6ffc92c43369/sdk/go/common/workspace/plugins.go#L1837